test/AccessControlTests/APITests.cpp

```diff
@@ -0,0 +1,171 @@
+                if (m_processHandle.is_valid())
+                {
+                    VERIFY_IS_TRUE(wil::handle_wait(m_processHandle.get(), 10000));
+
+                    DWORD exitCode{};
+                    VERIFY_WIN32_BOOL_SUCCEEDED(GetExitCodeProcess(m_processHandle.get(), &exitCode));
+                    VERIFY_ARE_EQUAL(exitCode, 0);
+                }
+                // Remove in reverse order to avoid conflicts between inter-dependent packages.
+                ::Test::Packages::RemovePackage(GetTestPackageFamilyName());
+                ::Test::Bootstrap::Cleanup();
+            }
+            catch (...)
+            {
+                return false;
+            }
+            return true;
+        }
+
+        TEST_METHOD_SETUP(MethodInit)
+        {
+            VERIFY_IS_TRUE(::Test::Packages::IsPackageRegistered_WindowsAppRuntimeFramework());
+            VERIFY_IS_TRUE(::Test::Packages::IsPackageRegistered_DynamicDependencyDataStore());
+            VERIFY_IS_TRUE(::Test::Packages::IsPackageRegistered_DynamicDependencyLifetimeManager());
+            return true;
+        }
+
+        TEST_METHOD_CLEANUP(MethodUninit)
+        {
+            if (m_processHandle.is_valid())
+            {
+                VERIFY_IS_TRUE(wil::handle_wait(m_processHandle.get(), 10000));
+                DWORD exitCode{};
+                VERIFY_WIN32_BOOL_SUCCEEDED(GetExitCodeProcess(m_processHandle.get(), &exitCode));
+                VERIFY_ARE_EQUAL(exitCode, 0);
+            }
+            VERIFY_IS_TRUE(::Test::Packages::IsPackageRegistered_WindowsAppRuntimeFramework());
+            VERIFY_IS_TRUE(::Test::Packages::IsPackageRegistered_DynamicDependencyDataStore());
+            VERIFY_IS_TRUE(::Test::Packages::IsPackageRegistered_DynamicDependencyLifetimeManager());
+
+            m_processHandle.reset();
+            return true;
+        }
+
+        void WaitForEvent(void* securityDescriptor)
+        {
+            wil::unique_event win32_event;
+            SECURITY_ATTRIBUTES sa{ sizeof(sa), securityDescriptor, FALSE };
+            win32_event.create(wil::EventOptions::None, L"AccessControlTest_Event", &sa);
+
+            VERIFY_IS_TRUE(win32_event.wait(5000));
+        }
+
+        TEST_METHOD(FlatAPITest)
+        {
+            RunTestApp();
+
+            ::AppContainerNameAndAccess ac[1] = {{GetTestPackageFile(), EVENT_MODIFY_STATE | SYNCHRONIZE}};
+            wil::unique_hlocal_security_descriptor sd;
+            THROW_IF_FAILED(::GetSecurityDescriptorForAppContainerNames(1, ac, nullptr, EVENT_MODIFY_STATE | SYNCHRONIZE, &sd, nullptr));
+            WaitForEvent(sd.get());
+        }
+
+        TEST_METHOD(WinRTStringTest)
+        {
+            RunTestApp();
+            winrt::AppContainerNameAndAccess ac[1] = { {GetTestPackageFile(), EVENT_MODIFY_STATE | SYNCHRONIZE} };
+
+            auto sdString = winrt::SecurityDescriptorHelpers::GetSddlForAppContainerNames(ac, L"", EVENT_MODIFY_STATE | SYNCHRONIZE);
```

Comment:
This is an instance where this call is going to generate an extra allocation for the hstring in the `ac` struct.